tests/integration/test_rest_catalog.py

```diff
@@ -16,34 +16,788 @@
+                {"id": 2, "name": "bar", "type": "int", "required": True},
+                {"id": 3, "name": "baz", "type": "boolean", "required": False},
+            ),
+            "schema-id": 0,
+            "identifier-field-ids": [2],
+        }
+    ],
+    "partition-specs": [
+        {"spec-id": 0, "fields": ({"source-id": 1, "field-id": 1000, "transform": "truncate[3]", "name": "id"},)}
+    ],
+    "default-spec-id": 0,
+    "last-partition-id": 1000,
+    "default-sort-order-id": 1,
+    "sort-orders": [
+        {
+            "order-id": 1,
+            "fields": [
+                {"source-id": 2, "transform": "identity", "direction": SortDirection.ASC, "null-order": NullOrder.NULLS_FIRST}
+            ],
+        }
+    ],
+    "properties": {
+        "owner": "fokko",
+        "write.parquet.compression-codec": "zstd",
+    },
+    "refs": {},
+    "snapshots": [],
+    "snapshot-log": [],
+    "metadata-log": [],
+}
+
+
+@pytest.fixture
+def table_metadata_no_snapshot_v2() -> Dict[str, Any]:
+    return EXAMPLE_table_metadata_no_snapshot_v2
+
 
-TEST_NAMESPACE_IDENTIFIER = "TEST NS"
+@pytest.fixture
+def table_metadata_no_snapshot_partitioned_v2() -> Dict[str, Any]:
+    return EXAMPLE_table_metadata_no_snapshot_partitioned_v2
+
+
+@pytest.fixture
+def rest_integration_example_metadata_partitioned_v2(table_metadata_no_snapshot_partitioned_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata-location": f"s3://warehouse/{TEST_TABLE_IDENTIFIER[0]}/{TEST_TABLE_IDENTIFIER[1]}",
+        "metadata": table_metadata_no_snapshot_partitioned_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def example_table_metadata_with_no_location(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def table_metadata_no_snapshot_rest(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata-location": f"s3://warehouse/{TEST_TABLE_IDENTIFIER[0]}/{TEST_TABLE_IDENTIFIER[1]}",
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
 
 
 @pytest.mark.integration
-@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog")])
-def test_namespace_exists(catalog: RestCatalog) -> None:
-    if not catalog.namespace_exists(TEST_NAMESPACE_IDENTIFIER):
+@pytest.fixture(scope="function")
+@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up")])
```

Comment:
nit: you can use something like this to automatically run clean up after every test 
https://stackoverflow.com/questions/22627659/run-code-before-and-after-each-test-in-py-test

```
@pytest.fixture(autouse=True)
def cleanup():
    # Code that will run before your test, for example:
    yield
    # Code that will run after your test, for example:
```